apm-agent-core/src/main/java/co/elastic/apm/agent/report/AbstractIntakeApiHandler.java

```diff
@@ -156,18 +166,36 @@ public void endRequest() {
                 try {
                     onRequestError(connection.getResponseCode(), connection.getErrorStream(), e);
                 } catch (IOException e1) {
-                    onRequestError(-1, connection.getErrorStream(), e);
+                    long maxRequestTime = TimeUnit.MILLISECONDS.toNanos(reporterConfiguration.getApiRequestTime().getMillis());
+                    long significantlyOverApiRequestTime = requestStartedNanos + maxRequestTime + TimeUnit.SECONDS.toNanos(1);
+                    // if the request has been open for significantly longer than api_request_time
+                    // it's quite likely that APM Server or a proxy has already closed the connection
+                    // we don't want to trigger a backoff or log an error if we have already flushed all data previously
+                    // note: flushing the OutputStream of a closed connection does not necessarily lead to an exception.
+                    // only when closing the connection, the exception occurrs
+                    // use case: on AWS Lambda, we flush after every request
+                    // and the environment may freeze for a long period of time so that the APM Server closes the connection
+                    if (!pendingFlush || clock.nanoTicks() >= significantlyOverApiRequestTime) {
+                        onRequestSuccess();
+                    } else {
+                        onRequestError(-1, connection.getErrorStream(), e);
+                    }
```

Comment:
This contains multiple heuristics:
1. that the `api_request_time` value is close to the `read_timeout` setting in the APM server
2. that one second is "significantly over API request time"
3. that crossing this significantly higher-than-request-duration can be considered as a successful request

While we can certainly live with heuristics if needed, since there is no real use for this functionality at the moment, I suggest we leave it out.

We can still consider leaving the `softFlush` functionality (without assuming freezes), but I don't see a real use case for that at the moment, especially since there are no guarantees that the flushed bytes are also flushed from the local buffer.